aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1437,14 +1440,104 @@ impl AptosVM {
 
     fn process_validator_transaction(
         &self,
-        _resolver: &impl AptosMoveResolver,
-        _txn: ValidatorTransaction,
-        _log_context: &AdapterLogSchema,
-    ) -> (VMStatus, VMOutput) {
-        (
-            VMStatus::Executed,
-            VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
-        )
+        resolver: &impl AptosMoveResolver,
+        txn: ValidatorTransaction,
+        log_context: &AdapterLogSchema,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        let session_id = SessionId::validator_txn(&txn);
+        match txn {
+            ValidatorTransaction::ObservedJWKsUpdates { updates } => {
+                self.process_jwk_updates(resolver, updates, log_context, session_id)
+            },
+            ValidatorTransaction::DummyTopic1(_) => Ok((
+                VMStatus::Executed,
+                VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
+            )),
+            ValidatorTransaction::DummyTopic2(_) => Ok((
+                VMStatus::Executed,
+                VMOutput::empty_with_status(TransactionStatus::Keep(ExecutionStatus::Success)),
+            )),
+        }
+    }
+
+    fn process_jwk_updates(
+        &self,
+        resolver: &impl AptosMoveResolver,
+        qc_updates: Vec<QuorumCertifiedUpdate>,
+        log_context: &AdapterLogSchema,
+        session_id: SessionId,
+    ) -> Result<(VMStatus, VMOutput), VMStatus> {
+        info!("[JWK] executing qc updates, qc_updates={:?}", qc_updates);
+
+        let validator_set = load_on_chain_config_from_resolver::<ValidatorSet>(resolver)
+            .map_err(|e| {
+                VMStatus::error(
+                    StatusCode::INVALID_SIGNATURE,
```

Comment:
these error codes are just placeholders. What should be the actual value?

Reply:
@vgao1996 Can you give some advice here?

Reply:
Note: we should propagate storage error when we call `get_resource` inside `load_on_chain_config_from_resolver` instead of using anyhow